### PR TITLE
documentation #767: Updated swagger documentation for auth endpoints

### LINKED ISSUE #767: Update Swagger doc for Auth (1)
We are updating the documentation for all endpoints and adding more _description and details_. 

One of the main controllers is Auth. You should update the documentation according to the **pattern** below:
```ts
  @ApiBearerAuth()
  @ApiOkResponse({
    type: DisciplineTeachersResponse,
  })
  @ApiBadRequestResponse({
    description: `\n
    InvalidQueryException:
      Type of discipline must be a field of enum
    InvalidDisciplineIdException:
      Discipline with such id is not found`,
  })
  @ApiUnauthorizedResponse({
    description: `\n
    UnauthorizedException:
      Unauthorized`,
  })
  @ApiForbiddenResponse({
    description: `\n
    NoPermissionException:
      You do not have permission to perform this action`,
  })
  @ApiParam({
    name: 'disciplineId',
    required: true,
    description: 'Id of certain discipline',
  })
  @ApiQuery({
    name: 'disciplineType',
    required: true,
    enum: DisciplineTypeEnum,
    description: 'Discipline type of some discipline',
  })
  @Access(PERMISSION.GROUPS_$GROUPID_DISCIPLINES_TEACHERS_GET, GroupByDisciplineGuard)
  @Get('/:disciplineId/teachers')
  async getAllByDiscipline (
    @Param('disciplineId', DisciplineByIdPipe) disciplineId: string,
    @Query('disciplineType') disciplineType: DisciplineTypeEnum
  ) {
    const teachers = await this.disciplineService.getTeachers(disciplineId, disciplineType);
    return { teachers };
  }
```
**Structure:**
- auth (if needed)
- ok response
- client errors
- params/query description
- access (if needed)

You should also add **field description** in the bady or response.
Add **summary** to endpoints (fix this issue).
Set **default values** for the optional fields

image

_My DM is always open for you. Happy coding_ 🐸

[Our Swagger](https://apidev.fictadvisor.com/api)
[Doc](https://docs.nestjs.com/openapi/introduction)

### PR DESCRIPTION
**Pull request by PavloVlasiuk**
- updated login
- updated registerTelegram
- updated register
- updated loginTelegram
- updated refresh
- updated updatePassword

closes #767

### CODE REVIEW

File: src/v2/api/dtos/RegisterTelegramDTO.ts [hunk 1]
```diff
@@ -2,11 +2,15 @@ import { IsNotEmpty } from 'class-validator';
 import { ApiProperty } from '@nestjs/swagger';
 
 export class RegisterTelegramDTO {
-  @ApiProperty()
+  @ApiProperty({
+    description: 'Token to verify user\'s telegram',
+  })
   @IsNotEmpty()
```
Comment (ArtemLyah) [inline src/v2/api/dtos/RegisterTelegramDTO.ts:8]: add IsString validation

File: src/v2/api/dtos/RegisterTelegramDTO.ts [hunk 2]
```diff
@@ -2,11 +2,15 @@ import { IsNotEmpty } from 'class-validator';
...
     token: string;
 
-  @ApiProperty()
+  @ApiProperty({
+    description: 'User\'s telegram id',
+  })
   @IsNotEmpty()
```
Comment (ArtemLyah) [inline src/v2/api/dtos/RegisterTelegramDTO.ts:14]: add IsNumber validation

File: src/v2/api/dtos/StudentDTO.ts [hunk 3]
```diff
@@ -3,32 +3,42 @@ import { UKR_REGEX, UKRSPEC_REGEX, validationOptionsMsg } from '../../utils/GLOB
 import { ApiProperty, ApiPropertyOptional } from '@nestjs/swagger';
 
 export class StudentDTO {
-  @ApiProperty()
+  @ApiProperty({
+    description: 'Id of the group',
+  })
   @IsNotEmpty(validationOptionsMsg('Group id can not be empty'))
```
Comment (ArtemLyah) [inline src/v2/api/dtos/StudentDTO.ts:9]: add IsString validation

File: src/v2/api/dtos/TelegramDTO.ts [hunk 4]
```diff
@@ -3,31 +3,45 @@ import { validationOptionsMsg } from '../../utils/GLOBALS';
...
+  })
   @IsNumber()
     id: bigint;
 
-  @ApiPropertyOptional()
+  @ApiPropertyOptional({
+    description: 'User\'s telegram lastName',
+  })
   @IsOptional()
     last_name: string;
```
Comment (ArtemLyah) [inline src/v2/api/dtos/TelegramDTO.ts:34]: write `last_name?: string;` if it's optional

File: src/v2/api/dtos/TelegramDTO.ts [hunk 5]
```diff
@@ -3,31 +3,45 @@ import { validationOptionsMsg } from '../../utils/GLOBALS';
...
 
-  @ApiPropertyOptional()
+  @ApiPropertyOptional({
+    description: 'Link to user\'s telegram photo',
+  })
   @IsOptional()
```
Comment (ArtemLyah) [inline src/v2/api/dtos/TelegramDTO.ts:39]: write `photo_url: string;` if it's optional

File: src/v2/api/responses/AuthLoginResponse.ts [hunk 6]
```diff
@@ -2,6 +2,8 @@ import { ApiProperty } from '@nestjs/swagger';
 import { AuthRefreshResponse } from './AuthRefreshResponse';
 
 export class AuthLoginResponse extends AuthRefreshResponse {
-  @ApiProperty()
+  @ApiProperty({
+    description: 'A credential to obtain new access token',
+  })
     refreshToken: string;
```
Comment (ArtemLyah) [inline src/v2/api/responses/AuthLoginResponse.ts:8]: add IsString validation

File: src/v2/api/responses/AuthRefreshResponse.ts [hunk 7]
```diff
@@ -1,6 +1,8 @@
 import { ApiProperty } from '@nestjs/swagger';
 
 export class AuthRefreshResponse {
-  @ApiProperty()
+  @ApiProperty({
+    description: 'A credential to access to protected resourses',
+  })
     accessToken: string;
```
Comment (ArtemLyah) [inline src/v2/api/responses/AuthRefreshResponse.ts:7]: add IsString validation

File: src/v2/api/dtos/StudentDTO.ts [hunk 8]
```diff
@@ -3,32 +3,42 @@ import { UKR_REGEX, UKRSPEC_REGEX, validationOptionsMsg } from '../../utils/GLOB
...
     new RegExp('^[' + UKR_REGEX + UKRSPEC_REGEX + ']{2,40}$'),
     validationOptionsMsg('Last name is not correct (A-Я(укр.)\\-\' ), or too short (min: 2), or too long (max: 40)'))
   @IsNotEmpty(validationOptionsMsg('Last name is empty'))
     lastName: string;
 
-  @ApiProperty()
+  @ApiProperty({
+    description: 'Whether the student is the head of the group or not',
+  })
   @IsBoolean()
```
Comment (ArtemLyah) [inline src/v2/api/dtos/StudentDTO.ts:42]: add validation option message like 'isCaptain is not boolean'

File: src/v2/api/dtos/RegisterTelegramDTO.ts (hunk 2 above)
Comment (ArtemLyah) [inline src/v2/api/dtos/RegisterTelegramDTO.ts:14]: add validation message like 'telegramId  cannot be empty'

File: src/v2/api/dtos/RegisterTelegramDTO.ts (hunk 1 above)
Comment (ArtemLyah) [inline src/v2/api/dtos/RegisterTelegramDTO.ts:8]: add validation message like 'token cannot be empty'

File: src/v2/api/controllers/AuthController.ts [hunk 9]
```diff
@@ -46,24 +46,37 @@ export class AuthController {
...
 
   @ApiBearerAuth()
   @ApiOkResponse()
-  @UseGuards(TelegramGuard)
+  @ApiUnauthorizedResponse({
+    description: `\n
+    UnauthorizedException:
+      Unauthorized`,
+  })
+  @ApiEndpoint({
```
Comment (ArtemLyah) [inline src/v2/api/controllers/AuthController.ts:76]: add ApiForbiddenResponse like this
`ts @ApiForbiddenResponse({
    description: `\n
    NoPermissionException:
      You do not have permission to perform this action`,
  })`

Comment (ArtemLyah) [inline src/v2/api/controllers/AuthController.ts:76]: add ApiForbiddenResponse like this
```ts
@ApiForbiddenResponse({
    description: `\n
    NoPermissionException:
      You do not have permission to perform this action`,
  })
```

File: src/v2/api/controllers/AuthController.ts [hunk 10]
```diff
@@ -46,24 +46,37 @@ export class AuthController {
...
+  })
   @Post('/login')
   async login (@Request() req) {
     return this.authService.login(req.user);
   }
 
   @ApiBearerAuth()
   @ApiOkResponse()
-  @UseGuards(TelegramGuard)
+  @ApiUnauthorizedResponse({
```
Comment (ArtemLyah) [inline src/v2/api/controllers/AuthController.ts:71]: add ApiBadRequestResponse with these validation messages
`Token cannot be empty`
`Telegram id cannot be empty`

**Review by ArtemiiLiakh — APPROVED** [review]

File: src/v2/api/controllers/AuthController.ts [hunk 11]
```diff
@@ -46,24 +46,43 @@ export class AuthController {
...
   })
-  @UseGuards(LocalAuthGuard)
+  @ApiEndpoint({
+    summary: 'Login to the user account',
+    guards: LocalAuthGuard,
+  })
   @Post('/login')
   async login (@Request() req) {
     return this.authService.login(req.user);
   }
```
Comment (Ksrebniukk) [inline src/v2/api/controllers/AuthController.ts:67]: there should be a body in this endpoint

Comment (Ksrebniukk) [reply src/v2/api/controllers/AuthController.ts:76]: there should be a body in this endpoint

**Review by fokaaas — APPROVED** [review]

**Pull request merged**